### PR TITLE
Lemon cipher

### PR DESCRIPTION
**Pull request by dennistreysa**
The Lemon Cipher is somewhat unknown. It originates from a letter sent to Franklin D. Roosevelt: http://scienceblogs.de/klausis-krypto-kolumne/2015/06/22/geloest-der-verschluesselte-drohbrief-an-us-praesident-roosevelt/

I don't know the proper name, but one day I got introduced to it as "Lemon Cipher".
The Clockwise/Counterclockwise option was an idea by myself.

There exists a tool to encrypt/decrypt: https://www.kryptographiespielplatz.de/index.php?aG=004a42fdaff92112591b4cac0e611ab0a648fe27

### CODE REVIEW

**Comment by dennistreysa** [comment]
I just realized the tool from Marcel Brätz works different then my tool for strings with odd length.

So for example `ABCDE` encrypts to `AEBDC`.
Which is wrong in my opinion.

I gues he initially does this like:

```
AE
BD
C
```

Which.... now that I look at it.... is exactly what I do in the "counterclockwise" mode.

I guess thats the reason I implemented both versions in the Multisolver back in 2015 ;)

File: lib/logic/tools/crypto/lemon.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import 'dart:math' as math;
+import 'package:gc_wizard/logic/tools/crypto/rotator.dart';
+import 'package:gc_wizard/utils/common_utils.dart';
+
+String encryptLemon(String input, {bool counterClockWise: true}) {
```
Comment (S-Man42) [inline lib/logic/tools/crypto/lemon.dart]: I would reuse a more global function, like the skytale function (maybe extract the skytale as global transposition function):

```
encrpytLemon{
   input = secondHalfOfInput.reverse + firstHalf
   return encryptSkytale(input, columnCount: input.length, countLettersPerCell: 1)
}
```

Just define what to do with odd number of input length.

File: assets/i18n/de.json [hunk 2]
```diff
@@ -393,5 +393,10 @@
   "enigma_numberrotors" : "Anzahl",
   "enigma_input" : "Eingabe",
 
-  "trithemius_title" : "Trithemius Chiffre"
+  "trithemius_title" : "Trithemius Chiffre",
+
+  "lemon_title" : "Lemon Chiffre",
+  "lemon_mode" : "Richtung",
+  "lemon_clockwise" : "UZS",
```
Comment (S-Man42) [inline assets/i18n/de.json]: Is "UZS" a common term? If you hadn't read "CW" before, I am not sure if I knew what that should mean :D

File: lib/widgets/registry.dart [hunk 3]
```diff
@@ -118,6 +119,7 @@ class Registry {
       GCWToolWidget(tool: Enigma(), toolName: i18n(context, 'enigma_title'), searchStrings: 'enigma rotors walzen'),
       GCWToolWidget(tool: KennysCode(), toolName: i18n(context, 'kennyscode_title'), searchStrings: 'they killed kenny sie haben kenny getötet kennys kenny\'s code southpark'),
       GCWToolWidget(tool: Trithemius(), toolName: i18n(context, 'trithemius_title'), searchStrings: SEARCHSTRING_VIGENERE + 'trithemius tabula recta'),
+      GCWToolWidget(tool: Lemon(), toolName: i18n(context, 'lemon_title'), searchStrings: 'lemon franklin roosevelt'),
```
Comment (S-Man42) [inline lib/widgets/registry.dart]: maybe +transposition?

File: lib/widgets/tools/crypto/lemon.dart [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        _buildOutput(context)
+      ],
+    );
+  }
+
+  Widget _buildOutput(BuildContext context) {
+    var output = '';
+
+    if (_currentMode == GCWSwitchPosition.left) {
+      output = encryptLemon(_currentInput, counterClockWise: (_direction != GCWSwitchPosition.left));
```
Comment (S-Man42) [inline lib/widgets/tools/crypto/lemon.dart]: I would prefer positive statements whenever possible: _direction == GCWSwitchPosition.right. Is more readable, I believe

**Comment by S-Man42** [comment]
Close this because of conflicting state and unsolved questions.

Would love to see this in the GCW, nonetheless. Please feel free to reopen this again if you like!

**Pull request closed without merging**